src/cript/api/api.py

```diff
@@ -38,52 +38,20 @@ def _get_global_cached_api():
     return _global_cached_api
 
 
-def _prepare_host(host: str) -> str:
-    """
-    prepares the host and gets it ready to be used within the api client
-
-    1. removes any trailing spaces from the left or right side
-    1. removes "/" from the end so that it is always uniform
-    1. adds "/api", so all queries are sent directly to the API
-
-    Parameters
-    ----------
-    host: str
-        api host
-
-    Returns
-    -------
-    host: str
-    """
-    # strip any empty spaces on left or right
-    host = host.strip()
-
-    # strip ending slash to make host always uniform
-    host = host.rstrip("/")
-
-    host = f"{host}/api/v1"
-
-    # if host is using unsafe "http://" then give a warning
-    if host.startswith("http://"):
-        warnings.warn("HTTP is an unsafe protocol please consider using HTTPS.")
-
-    if not host.startswith("http"):
-        raise InvalidHostError("The host must start with http or https")
-
-    return host
-
-
 class API:
     """
     ## Definition
     API Client class to communicate with the CRIPT API
     """
 
-    _host: str = ""
+    _raw_host: str = ""
+    _https_warning_sent: bool = False
     _token: str = ""
     _vocabulary: dict = {}
     _db_schema: dict = {}
     _http_headers: dict = {}
+    _api_handle: str = "api"
+    _api_version: str = "v1"
```

Comment:
is there anyway that we can encapsulate everything needed for the api host into one or two function, so nothing else needs to have access to it because I don't think anything else needs it

Reply:
This is encapsulated.
These are private memebers of the class. That is about as encapsulated as you can get.

Reply:
we could encapsulate them within a function and not have the rest of the class see it or need it, we only set the host variable for the class and the rest of the class never sees this